opencog/atomspace/AtomSpace.h

```diff
@@ -358,6 +374,57 @@ class AtomSpace
     }
 
     /**
+     * Gets all the atoms of any type and appends them to the HandleSeq.
+     *
+     * @param appendToHandles the HandleSeq to which to append the handles.
+     *
+     * Example of call to this method:
+     * @code
+     *         std::list<Handle> atoms;
+     *         atomSpace.get_all_atoms(atoms);
+     * @endcode
+     */
+    void get_all_atoms(HandleSeq& appendToHandles) const
+    {
+        // Defer to handles by type for ATOMs and subclasses.
+        get_handles_by_type(appendToHandles, ATOM, true);
```

Comment:
Oh, I figured it was historical. 

I myself should have gone through the code myself last year and removed the no-longer-necessary `back_iterator` wrappers when I added the optimization to `reserve` only once and made the non-iterator method much faster for large return sizes.